lms/djangoapps/lti_provider/views.py

```diff
@@ -173,7 +177,26 @@ def render_courseware(request, lti_params):
         'disable_footer': True,
         'disable_tabs': True,
         'staff_access': staff,
-        'xqa_server': settings.FEATURES.get('XQA_SERVER', 'http://your_xqa_server.com'),
+        'xqa_server': settings.FEATURES.get('XQA_SERVER', 'http://example.com/xqa'),
     }
 
     return render_to_response('courseware/courseware.html', context)
+
+
+def parse_course_and_usage_keys(course_id, usage_id):
+    """
+    Convert course and usage ID strings into key objects. Return a tuple of
+    (course_key, usage_key), or (None, None) if the translation fails.
+    """
+    try:
+        course_key = CourseKey.from_string(course_id)
+    except InvalidKeyError:
+        return None, None
+    if not course_key:
+        return None, None
+    try:
+        usage_id = unquote_slashes(usage_id)
+        usage_key = UsageKey.from_string(usage_id).map_into_course(course_key)
+    except InvalidKeyError:
+        return None, None
```

Comment:
Maybe just let the `InvalidKeyError` bubble up instead of returning `None`? Diapering exceptions into `None`s tends to trip up people eventually, masking where the real problem is when they try to use one of these keys far away from where they parsed it.